.pre-commit-hooks.yaml

```diff
@@ -0,0 +1,6 @@
+-   id: byexample
+    name: run code snippets in docstrings to validate them
+    description: avoids writing errorneous code snippets.
```

Comment:
While it is correct that a failing example may mean an error in the code snippet, it is only one side of the coin. A failing example may mean that the snippet is correct but the source code that you are documenting is not.

Perhaps a better description could be:

```
byexample a literate programming engine where you mix ordinary text and snippets of code in the same file and then you execute them as regression tests.

You can always be sure that the examples are correct and your documentation is up to date!
```

I couldn't find where this information is shown or used by `pre-commit` but I guess that a multiline description like above should be supported.

Reply:
Well I did not find much information about this neither. I mostly got inspiration from pre-commit’s [pre-commit-hooks](https://github.com/pre-commit/pre-commit-hooks).
However, I don’t think the description matters much as it won’t be read by many people: in my opinion, people that use per-commit hooks already know what the hooks are doing or should get more information elsewhere than in that very file.